spi/src/main/java/com/flipkart/grayskull/spi/authz/AuthorizationContext.java

```diff
@@ -0,0 +1,75 @@
+package com.flipkart.grayskull.spi.authz;
+
+import com.flipkart.grayskull.models.db.Project;
+import com.flipkart.grayskull.models.db.Secret;
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.ToString;
+import org.springframework.security.core.Authentication;
+
+import java.util.Optional;
+
+/**
+ * An immutable container for all contextual information required to make an authorization decision.
+ * It encapsulates the authenticated principal, the project being accessed, and optionally, the specific secret.
+ */
+@AllArgsConstructor(access = AccessLevel.PRIVATE)
+@Getter
+@ToString
+public final class AuthorizationContext {
+
+    /** The authenticated principal and their credentials. */
+    private final Authentication authentication;
+
+    /** The project scope for the authorization check. */
+    private final Project project;
```

Comment:
philosophical question - why should grayskull.spi.authz package depend on grayskul.models packages? Should we remove the direct reference of these objects and make them primitive data types? you only really need the projetId string and secret name string here